### ISSUE TITLE
bar-ui, how to refresh playlistController elements after changing DOM playlist?

### ISSUE THREAD

**Issue by dessalines** [scottschiller/SoundManager2#121]
So I'm using [jquery-sortable](http://jqueryui.com/resources/demos/sortable/placeholder.html) with bar-ui to be able to rearrange items in the playlist(Pretty neat huh, being able to drag and drop songs in that playlist container). The rearranging is working well, but after I've moved an item, lets say to play next, it skips over it.

I try doing `playlistController.refresh()`, but it doesn't fix the problem. Oddly, if I click another song, which I think runs the [playLink](https://github.com/scottschiller/SoundManager2/blob/master/demo/bar-ui/script/bar-ui.js#L337) function, it **does** update the playlist and fixes the problem. I've tried to go through that function, but can't figure out what its doing differently that causes it to update from the DOM.

Basically, I need a programmatic way to update the playlistController from the DOM, because playlistController.refresh() doesn't seem to work.

**Comment by zirafa** [comment]
My experience is unrelated to bar-ui, but I encountered this sort of odd behavior when working on the [autoscan + jquery-sortable demo for pushtape-player.js](http://zirafa.github.io/pushtape-player.js/demo/autoscan-example.html). The jquery-sortable plugin is somewhat finicky about when it fires certain events. It's possible you are calling playlistController.refresh() in the wrong place or at the wrong time. Calling it inside jquery-sortable's [event-update](http://api.jqueryui.com/sortable/#event-update) *should* work, however. Adding some debug code there may help shed some light on the issue.

Incidentally, I solved this in pushtape-player.js in a completely different way without needing to fire refresh events. I use a mutationobserver event attached to the playlist container, which automatically observes any DOM changes and scans/updates the internal playlist index as needed.
